updatehub-sdk/src/listener.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2018, 2019, 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Result;
+use async_ctrlc::CtrlC;
+use async_std::{
+    io::BufReader,
+    os::unix::net::{UnixListener, UnixStream},
+    prelude::*,
+};
+use std::{collections::HashMap, net::Shutdown, process};
+
+const SDK_TRIGGER_FILENAME: &str =
+    "/usr/share/updatehub/state-change-callbacks.d/10-updatehub-sdk-statechange-trigger";
+const SOCKET_PATH: &str = "updatehub-statechange.sock";
+
+type StateChangeCallback = fn(State);
+type ErrorCallback = fn(String);
+
+pub struct StateChangeListener {
+    listeners: HashMap<String, StateChangeCallback>,
```

Comment:
On a second note, this type struct doesn't support multiple callbacks being registered for the same state. Is this limitation present in the others SDK as well? I'm fine either way as long as the behavior is consistent.